scipy/cluster/hierarchy.py

```diff
@@ -3558,8 +3572,8 @@ def _dendrogram_calculate_info(Z, p, truncate_mode,
     # !!! Otherwise, we don't have a leaf node, so work on plotting a
     # non-leaf node.
     # Actual indices of a and b
-    aa = int(Z[i - n, 0])
-    ab = int(Z[i - n, 1])
+    aa = xp.astype(Z[i - n, 0], xp.int64)
```

Comment:
This pattern occurs a lot. `int` should work I'd think - what am I missing? 

I tried with a GPU array quickly, and no problem there:
```python
>>> import torch
>>> t = torch.tensor(3, device='cuda')
>>> int(t)
3
>>> type(int(t))
<class 'int'>
```

Reply:
So, I guess there was something else I had to fix because now I could remove that. I just needed to keep the only float one because I had these issues otherwise:


```
# X = float(X)

X          = array([1])
E   DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)

y          = tensor(1.)
ValueError: `y` must be 1 or 2 dimensional.

X          = Array([1], dtype=int64)
TypeError: float is only allowed on arrays with 0 dimensions
```

Reply:
So, I guess there was something else I had to fix because now I could remove that. I just needed to keep the only float one because I had these issues otherwise:


```
# X = float(X)

X          = array([1])
E   DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)

y          = tensor(1.)
ValueError: `y` must be 1 or 2 dimensional.

X          = Array([1], dtype=int64)
TypeError: float is only allowed on arrays with 0 dimensions
```

Reply:
Great - this looks good now.